packages/starlight/components/SidebarSublist.astro

```diff
@@ -43,6 +47,22 @@ interface Props {
     margin-top: 0.75rem;
   }
 
+  .sidebar-group-header {
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    cursor: pointer;
```

Comment:
The designs we have for this have the padding around the full item:
<img width="554" alt="image" src="https://github.com/withastro/starlight/assets/357379/76b00ca9-86fa-47c9-a7b5-1b0db2cf7ce1">

We can add it here:

```suggestion
    padding-inline: var(--sl-sidebar-item-padding-inline);
    cursor: pointer;
```

We should also remove the `padding-inline` from `h2` in this case.